### ISSUE TITLE
[Cart] Line item error text not announced via assistive technology

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#471]
## Issue summary

when the user adjusts an item which hits a limit, an error message is displayed visually below the product information. However, this text content is not announced or conveyed to screen reader users. Blind or low-vision screen reader users would miss this context and may not understand why the quantity `input` was reset. This would result in a confusing or frustrating user experience.

<details>
  <summary>Screenshots</summary>

<img width="1039" alt="Dawn cart page, quantity error text is highlighted." src="https://user-images.githubusercontent.com/1392632/130667617-47e2e612-3e14-4d5a-9db5-c77c3a238e01.png">

</details>

#### HTML

```html
<tr class="cart-item" id="CartItem-1">
  <!-- Product details… -->

  <td class="cart-item__details">
    <!-- … -->
    <p class="cart-item__error" id="Line-item-error-1">
      <span class="cart-item__error-text">
        You can only add 1 of this item to your cart.
      </span>
      <!-- Icon… -->
    </p>
  </td>

  <!-- Price… -->

  <td class="cart-item__quantity">
    <label class="medium-down" for="Quantity-1">Quantity</label>
    <quantity-input class="quantity">
      <button class="quantity__button no-js-hidden" name="minus" type="button">
        <span class="visually-hidden">Decrease quantity for Cydney Plaid</span>
        <!-- Icon… -->
      </button>
      <input
        class="quantity__input"
        type="number"
        name="updates[]"
        value="1"
        min="0"
        aria-label="Quantity for Cydney Plaid"
        id="Quantity-1"
        data-index="1"
      />
      <button class="quantity__button no-js-hidden" name="plus" type="button">
        <span class="visually-hidden">Increase quantity for Cydney Plaid</span>
        <!-- Icon… -->
      </button>
    </quantity-input>
  </td>

  <!-- Total/remove… -->
</tr>
```

## Steps to reproduce

1. Go to a Dawn store with any supported browser
2. Add an item to the cart **with only 1 inventory** and proceed to the cart page
3. With an active screen reader, increase the quantity of the item by 1

### Expected

- Error text to be announced aloud to inform the screen reader user of the current error context.

### Actual

- Error text is not announced.

### HTML structure…

1. Add the `role` attribute to the `.cart-item__error` content container, setting its value to `status`. This will create a live region which will announce dynamically injected text.
2. Adjust semantics:

- Swap `p.cart-item__error` for a `div` element
- Swap `span.visually-hidden` for a `small` element

#### HTML

```html
<tr class="cart-item" id="CartItem-1">
  <!-- Product details… -->

  <td class="cart-item__details">
    <!-- … -->
  </td>

  <!-- Price… -->

  <td class="cart-item__quantity">
    <label class="medium-down" for="Quantity-1">Quantity</label>
    <quantity-input class="quantity">
      <button class="quantity__button no-js-hidden" name="minus" type="button">
        <span class="visually-hidden">Decrease quantity for Cydney Plaid</span>
        <!-- Icon… -->
      </button>
      <input
        class="quantity__input"
        type="number"
        name="updates[]"
        value="1"
        min="0"
        aria-label="Quantity for Cydney Plaid"
        id="Quantity-1"
        data-index="1"
      />
      <button class="quantity__button no-js-hidden" name="plus" type="button">
        <span class="visually-hidden">Increase quantity for Cydney Plaid</span>
        <!-- Icon… -->
      </button>
    </quantity-input>
    <div class="cart-item__error" id="Line-item-error-1" role="status">
      <small class="cart-item__error-text">
        You can only add 1 of this item to your cart.
      </small>
      <!-- Icon… -->
    </div>
  </td>

  <!-- Total/remove… -->
</tr>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **WCAG Principal:** Robust
- **WCAG SC:** [4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html)
- **Severity:** High
- **Effort:** Low
- **Related:** https://github.com/Shopify/dawn/issues/470
